IdentityCore/src/network/MSIDHttpRequest.m

```diff
@@ -69,56 +70,59 @@ - (void)sendWithBlock:(MSIDHttpRequestDidCompleteBlock)completionBlock
     self.urlRequest = [self.requestSerializer serializeWithRequest:self.urlRequest parameters:self.parameters];
     
     [self.telemetry sendRequestEventWithId:self.context.telemetryRequestId];
+    [self.serverTelemetry setTelemetryToRequest:self];
```

Comment:
should this only be done for AAD http requests?

Reply:
MSIDHttpRequestServerTelemetryHandling is a general protocol, it is not AAD specific. In case if we don't need to send server telemetry, `serverTelemetry` property should be nil.